web-ui/src/app/components/address-details/address-details.component.ts

```diff
@@ -32,17 +39,30 @@ export class AddressDetailsComponent implements OnInit {
       response => this.onAddressRetrieved(response),
       response => this.onError(response)
     );
+    this.getPage(this.currentPage);
   }
 
-  private onAddressRetrieved(response: Address) {
+  private onAddressRetrieved(response: Balance) {
     this.address = response;
   }
 
+  getPage(page: number) {
+    const offset = (page - 1) * this.pageSize;
+    const limit = this.pageSize;
+    const order = 'time:desc';
+
+    this.asyncItems = this.addressesService
+      .getTransactions(this.addressString, offset, limit, order)
+      .do(response => this.total = response.total)
+      .do(response => this.currentPage = 1 + (response.offset / this.pageSize))
+      .map(response => response.data);
+  }
+
   private onError(response: any) {
     this.errorService.renderServerErrors(null, response);
   }
 
-  getSent(address: Address): number {
-    return address.received - address.balance;
+  getSent(address: Balance): number {
```

Comment:
this is not required because the `Balance` model has the `spent` attribute, before this was required because the model didn't have it.